### ISSUE TITLE
threadsafe

### ISSUE THREAD

**Issue by delanym** [rudikershaw/git-build-hook#31]
Pls mark as @threadSafe to support parallel building to avoid this warning

```
[WARNING] *****************************************************************                                                                                                                                   [WARNING] * Your build is requesting parallel execution, but project      *                                                                                                                                   
[WARNING] * contains the following plugin(s) that have goals not marked   *                                                                                                                                   
[WARNING] * as @threadSafe to support parallel building.                  *                                                                                                                                   
[WARNING] * While this /may/ work fine, please look for plugin updates    *                                                                                                                                   
[WARNING] * and/or request plugins be made thread-safe.                   *                                                                                                                                   
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in Traderoot Maven Super:
[WARNING] com.rudikershaw.gitbuildhook:git-build-hook-maven-plugin:3.1.0
[WARNING] Enable debug to see more precisely which goals are not marked @threadSafe.
[WARNING] *****************************************************************
```

**Comment by rudikershaw** [comment]
Hi @delanym,

I can certainly take a look to see what changes are required for me to confidentially mark it as thread safe. There are some goals for example that have file system operations that I would not be comfortable marking as thread safe.

**Comment by delanym** [comment]
@rudikershaw do you think the functionality provided in this plugin might be better suited to an extension with a config file?

**Comment by rudikershaw** [comment]
I have marked the initialize and configure goals as thread safe, and released the changes to 3.2.0. The install goal cannot be trivially made thread safe, and so it has not be marked as such.
